### PR TITLE
[script.playrandomvideos@matrix] 2.0.0

### PR DESCRIPTION
**Pull request by rmrector**
### Add-on details:

- **General**
  - Add-on name: Play Random Videos
  - Add-on ID: script.playrandomvideos
  - Version number: 2.0.0
  - Kodi/repository version: matrix

- **Code location**
  - URL: https://github.com/rmrector/script.playrandomvideos
  
This add-on can quickly play random episodes from TV shows, movies from genres/sets/years/tags, and videos from playlists, the filesystem, and just about anything else, other than plugins.

### Description of changes:

v2.0.0 (2021-01-29)
- Kodi 19 Matrix / Python 3 compatibility. Breaks compatibility with previous versions of Kodi.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: script.playrandomvideos/addon.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	</extension>
+	<extension point="xbmc.python.library" library="python/script.py" />
+	<extension point="xbmc.addon.metadata">
+		<summary lang="en_GB">Plays random videos from all sorts of lists.</summary>
+		<description lang="en_GB">This add-on can quickly play random episodes from TV shows, movies from genres/sets/years/tags, and videos from playlists, the filesystem, and just about anything else, other than plugins.</description>
+		<news>v2.0.0 (2021-01-29)
+- Kodi 19 Matrix / Python 3 compatibility. Breaks compatibility with previous versions of Kodi.</news>
+		<platform>all</platform>
+		<source>https://github.com/rmrector/script.playrandomvideos</source>
+		<forum>https://forum.kodi.tv/showthread.php?tid=238613</forum>
```
Comment (enen92) [inline script.playrandomvideos/addon.xml:26]: MIT (spdx identifier)

File: script.playrandomvideos/python/lib/player.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        if self.showbusydialog: busy.close()
+        if extended:
+            self.play(self.playlist)
+            if LOAD_NEW - 1 > 0:
+                self.extend_playlist(LOAD_NEW - 1)
+        while xbmc.getCondVisibility('Player.HasMedia'):
+            if self.source_exhausted:
+                break
+            xbmc.sleep(2000)
+
```
Comment (enen92) [inline script.playrandomvideos/python/lib/player.py:49]: Debug

File: script.playrandomvideos/python/lib/player.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        if self.source_exhausted:
+            return False
+        addedcount = 0
+        for _ in range(0, count):
+            try:
+                self.add_to_playlist(next(self._source))
+                addedcount += 1
+            except StopIteration:
+                self.source_exhausted = True
+                log("Source has been exhausted", xbmc.LOGINFO)
```
Comment (enen92) [inline script.playrandomvideos/python/lib/player.py:65]: Debug

**Review by enen92 — COMMENTED** [review]
A few minors and a clarification. Thanks for the update.

File: script.playrandomvideos/python/context.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import sys
+import xbmc
+
+from lib import playrandom
+from lib.pykodi import get_pathinfo
+
+if __name__ == '__main__':
+    path = sys.listitem.getPath()
```
Comment (enen92) [inline script.playrandomvideos/python/context.py:8]: Can you explain how the listitem gets into sys?

Comment (rmrector) [reply script.playrandomvideos/python/context.py:8]: Magic that Kodi does for context item add-ons. I'm not sure how it works internally but it has a quck mention in the wiki, section 2, https://kodi.wiki/view/Context_Item_Add-ons .

Comment (enen92) [reply script.playrandomvideos/python/context.py:8]: Thanks

**Review by enen92 — APPROVED** [review]

**Pull request merged**